### PR TITLE
fix(Core/Spell): Remove ErrorCube visual

### PR DESCRIPTION
**Pull request by Kitzunu**
* cherry-pick https://github.com/trinitycore/trinitycore/commit/3b5014fdae

* cherry-pick small part of https://github.com/TrinityCore/TrinityCore/commit/f4f7e6324d6c99335bf3479f212edea1e5572f88

Co-Authored-By: Shauren <shauren.trinity@gmail.com>
Co-Authored-By: Treeston <14020072+treeston@users.noreply.github.com>

## Changes Proposed:
- Load SpellVisual.dbc
-  Implement SPELL_ATTR0_CU_NEEDS_AMMO_DATA
-  Fix 60988 from being a cube when cast

## Tests Performed:

- Built
- Tested in-game

## How to Test the Changes:

1. .cast 60988
2. see before is a cube, after not

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Spells/SpellMgr.cpp [hunk 1]
```diff
@@ -3257,6 +3257,20 @@ void SpellMgr::LoadSpellCustomAttr()
                 spellInfo->Effects[EFFECT_0].MiscValue = 127;
                 break;
         }
+
+        if (spellInfo->Speed > 0.0f)
+        {
+            if (SpellVisualEntry const* spellVisual = sSpellVisualStore.LookupEntry(spellInfo->SpellVisual[0]))
+            {
+                if (spellVisual->HasMissile)
```
Comment (Winfidonarleyan) [inline src/server/game/Spells/SpellMgr.cpp:3265]: Maybe 1 condition?

**Review by Winfidonarleyan — APPROVED** [review]

File: src/server/game/Spells/SpellMgr.cpp (hunk 1 above)
Comment (Kitzunu) [reply src/server/game/Spells/SpellMgr.cpp:3265]: Do we want to combine where we define spellVisual, maybe it gets messy to read.

But I could do 
```cpp
if (spellVisual->HasMissile && (spellVisual->MissileModel == -4 || spellVisual->MissileModel == -5))
```
If you think that is good?

Comment (Kitzunu) [reply src/server/game/Spells/SpellMgr.cpp]: updated

**Comment by Branel** [comment]
The spell_visual_dbc does not return any affected rows when used on mysql, the other mysql works fine. Unable to test.

**Comment by Kitzunu** [comment]
Dont worry about the DB table, it wont be used unless for custom work.

Test the spell ingame

**Comment by Branel** [comment]
Then didnt work at all, made sure that I am on the right PR.

**Comment by Kitzunu** [comment]
Wtf? I am 100% sure this works

**Comment by Kitzunu** [comment]
I will test this again tonight

**Comment by Branel** [comment]
https://user-images.githubusercontent.com/41377234/118259364-5a2c5d00-b4b1-11eb-83b5-453281031183.mp4

**Comment by Kitzunu** [comment]
Well isn't this lovely
I promise it worked, but something new to master must have broken something... oh well

**Comment by Kitzunu** [comment]
I dont know anymore if someone has a clue pls tell me.

**Review by lineagedr — APPROVED** [review]

**Comment by Kitzunu** [comment]
Was tested by both me and Lineagedr in-game

**Pull request merged**